### ISSUE TITLE
Start DocApi development

### ISSUE THREAD

**Issue by Mec-iS** [HTTP-APIs/hydrus-real-world-app#1]
As the objective of this repo is to create an hydrus deployment with the characteristics defined by the ["Real World Example"](https://github.com/gothinkster/realworld), we start analyzing the views and models for the [Flask app](https://github.com/gothinkster/flask-realworld-example-app) to turn them into an Hydra documentation file.
The Example app is a basic Medium.com clone, the Flask deployment is called "Conduit"; _three main classes of objects_ are defined:
* Articles
* Profiles
* Users

### Articles
* SQLAlchemy models: [Tags Comments Articles](https://github.com/gothinkster/flask-realworld-example-app/blob/master/conduit/articles/models.py). These models have to be translated into `HydraClass`
* Views: [Articles Comments Tags](https://github.com/gothinkster/flask-realworld-example-app/blob/master/conduit/articles/views.py). The definitions of these views have to be added to the DocApi's `HydraClass`s

### Profiles
* Same as Articles, there are some models and views that needs to be translated

### Users
* Same as Articles, there are some models and views that needs to be translated

I will add an ApiDoc template to the repo via PR.

Best way to start is to find the relevant schemas at schema.org; for example an Article is probably an instance of class [BlogPosting](https://schema.org/BlogPosting).

**Comment by Mec-iS** [comment]
[Template](https://github.com/HTTP-APIs/hydrus-real-world-app/blob/master/RealWorldApiDoc.jsonld)

**Comment by vddesai1871** [comment]
Sounds like an interesting exercise! 

It would be great if we can **document this process** somewhere (including what you have written above, the process of finding relevant schema definitions, creation of ApiDoc, running hydrus with that ApiDoc and other details). To provide a go-to guide explaining *creation of HTTP-API with hydrus*. Will bridge the gap between various documentation pieces we have right now and provide a very useful guide for anyone new to hydrus or semantic web in general.

**Comment by knakul853** [comment]
@vddesai1871  sounds cool can we open issue for this?

**Comment by Mec-iS** [comment]
Possibly the first step is to map Real-World-App objects classes to schema.org classes

**Comment by vddesai1871** [comment]
>Possibly the first step is to map Real-World-App objects classes to schema.org classes

If I am not missing something,
1) The first part would be to identify all the classes as you mentioned in the issue description and see whether the identified class would be a collection class or not. 
2) Then we will have to map the properties of these class to schema.org classes
3) Then identify valid operations over each class and identify entrypoint details.

**Comment by Mec-iS** [comment]
> The first part would be to identify all the classes as you mentioned in the issue description and see whether the identified class would be a collection class or not.

Yeah, first is to write down the `HydraClass`s mapping the models and their properties. Then write down the `Hydra:Resource`s and `Hydra:Collection`s. Finally the `Hydra:Entrypoint` for all the resources/collections.

**Comment by gustavodemorais** [comment]
Studying the specification of the backend API at https://github.com/gothinkster/realworld/tree/master/api there is something I cannot get my head around yet:

- How to recover multiple Articles and not the entire collection according to filters, as the image below show:

![image](https://user-images.githubusercontent.com/11739405/56047038-dc7f4b80-5d44-11e9-8966-44b061e3f9de.png)

So far we don't have a way to process requests from what I understand.

**Comment by Mec-iS** [comment]
@Guttz Exactly, for that part we need to wait to integrate `IRItemplate` specs in hydrus.
In the meanwhile we can implement resources and collections without the filters.
